prboom2/src/dsda/tempdir.c

```diff
@@ -0,0 +1,185 @@
+//
+// Copyright(C) 2022 by Pierre Wendling
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// DESCRIPTION:
+//	DSDA get temporary directories
+//
+
+#include "config.h"
+
+#ifdef HAVE_MKDTEMP
+#if defined(HAVE_MKDTEMP_STDLIB)
+#if defined(HAVE_MKDTEMP_POSIX_SOURCE)
+#define _POSIX_C_SOURCE 200809L
+#elif defined(HAVE_MKDTEMP_BSD_SOURCE)
+#define _BSD_SOURCE
+#elif defined(HAVE_MKDTEMP_DEFAULT_SOURCE)
+#define _DEFAULT_SOURCE
+#endif
+#include <stdlib.h>
+#elif defined(HAVE_MKDTEMP_UNISTD)
+#include <unistd.h>
+#endif
+#endif
+
+#ifdef HAVE_GET_TEMP_PATH
+/* Mingw requires these two headers first */
+#include <tchar.h>
+#include <windows.h>
+
+#include <inttypes.h>
+#include <stdlib.h>
+#include <string.h>
+#include <strsafe.h>
+#if defined(HAVE_GET_TEMP_PATH_2)
+#define winapi_GetTempDir GetTempPath2
```

Comment:
We don't define UNICODE, so we must use the `GetTempPath2W`/`GetTempPathW` function to support non-Latin paths. Typically `%TEMP%` is found in the user's directory on Windows, with an often non-Latin name. I recommend just using `tempdir = M_getenv("TEMP");` it works for us in Woof/Crispy. See: See: https://github.com/chocolate-doom/chocolate-doom/blob/981da81f2adabd6eb82f1c3fc4ea27baecfccd82/src/m_misc.c#L572

Reply:
Heh, you're probably right. I'll just go ahead and use the existing constructs instead of making a whole new module.